src/components/SingleDatePicker.jsx

```diff
@@ -344,7 +620,8 @@ export default class SingleDatePicker extends React.Component {
   }
 
   isHovered(day) {
-    return isSameDay(day, this.state.hoverDate);
+    const { hoverDate } = this.state || {};
```

Comment:
:-/ hmm - in that case it kind of seems like the meat of it should be extracted out to a pure function in a closure, and have both the constructor and `isHovered` call into it with the data it needs.